jerry-core/ecma/operations/ecma-promise-object.c

```diff
@@ -227,14 +233,47 @@ ecma_reject_promise (ecma_value_t promise, /**< promise */
  *
  * See also: ES2015 25.4.1.4
  */
-static void
+void
 ecma_fulfill_promise (ecma_value_t promise, /**< promise */
                       ecma_value_t value) /**< fulfilled value */
 {
   ecma_object_t *obj_p = ecma_get_object_from_value (promise);
 
   JERRY_ASSERT (ecma_promise_get_flags (obj_p) & ECMA_PROMISE_IS_PENDING);
 
+  if (promise == value)
+  {
+    ecma_value_t exception = ecma_raise_type_error (ECMA_ERR_MSG ("A promise cannot be resolved with itself."));
+    JERRY_ASSERT (ECMA_IS_VALUE_ERROR (exception));
+
+    exception = jcontext_take_exception ();
```

Comment:
How about eliminating this double assignment?

```c
    ecma_raise_type_error (ECMA_ERR_MSG ("A promise cannot be resolved with itself."));
    ecma_value_t exception = jcontext_take_exception ();
```

Reply:
I did it for the assert. If it bothers you, I can remove it.

Reply:
I'd support removing it, otherwise the patch looks good to me.